src/core/drive/page_view.js

```diff
@@ -16,7 +16,7 @@ export class PageView extends View {
   }
 
   renderPage(snapshot, isPreview = false, willRender = true, visit) {
-    const shouldMorphPage = this.isPageRefresh(visit) && this.snapshot.shouldMorphPage
+    const shouldMorphPage = this.shouldMorphPage(snapshot, visit)
```

Comment:
Since the `shouldMorphPage(..)` is only used here, what about removing the `snapshot` argument and naming it more in relation to a visit? Eg. `isMorphableVisit(visit)` or something like that and keep the `this.snapshot` clause on this line. This is a bit closer to the original code as well and would remove the variable name being the same as the method name :)